src/withPseudoState.js

```diff
@@ -37,26 +37,37 @@ const shadowHosts = new Set()
 addons.getChannel().on(STORY_CHANGED, () => shadowHosts.clear())
 
 // Global decorator that rewrites stylesheets and applies classnames to render pseudo styles
-export const withPseudoState = (StoryFn) => {
-  const parameter = useParameter("pseudo")
+export const withPseudoState = (StoryFn, { viewMode, parameters, id }) => {
+  const isDocs = viewMode === 'docs';
+  const { pseudo: parameter } = parameters;
   const [{ pseudo: globals }, updateGlobals] = useGlobals()
 
-  // Sync parameter to globals, used by the toolbar
+  // Sync parameter to globals, used by the toolbar (only in canvas as this
+  // doesn't make sense for docs because many stories are displayed at once)
   useEffect(() => {
-    if (parameter !== globals) updateGlobals({ pseudo: parameter })
-  }, [parameter])
+    if (parameter !== globals && !isDocs) updateGlobals({ pseudo: parameter })
+  }, [parameter, isDocs])
 
   // Convert selected states to classnames and apply them to the story root element.
   // Then update each shadow host to redetermine its own pseudo classnames.
   useEffect(() => {
-    applyClasses(
-      document.getElementById("root"),
-      Object.entries(globals || {})
-        .filter(([_, value]) => value)
-        .map(([key]) => `pseudo-${PSEUDO_STATES[key]}`)
-    )
-    shadowHosts.forEach(updateShadowHost)
-  }, [globals])
+    const apply = element => {
+      applyClasses(
+        element,
+        Object.entries(parameter || {})
+          .filter(([_, value]) => value)
+          .map(([key]) => `pseudo-${PSEUDO_STATES[key]}`)
+      )
+      shadowHosts.forEach(updateShadowHost)
+    }
```

Comment:
I'd extract this function outside of the component and add `parameter = {}` as the second argument.